skimage/util/dtype.py

```diff
@@ -32,7 +32,68 @@
 _supported_types = list(dtype_range.keys())
 
 
-def dtype_limits(image, clip_negative=False):
+def _check_precision_loss(dtypeobj_in, dtypeobj_out):
+    """Check if conversion between images will incur loss of precision.
+
+    Generally speaking, the output object needs to have more significant bits
+    than the input kind.
+
+    Details:
+        float -> signed/unsigned ints will always return true.
+        signed/unsigned -> float of same number of bytes will return true.
+
+    Parameters
+    ----------
+    dtypeobj_in: np.dtype
+        dtype of the input image.
+    dtypeobj_out: np.dtype
+        dtype of the output image.
+    output_warning: bool, optional
+        Outputs a warning using `warn` if operation will incur loss of
+        precision.
+    int_same_size_ok: bool, optional
+        Should conversion between integers of the same same (and signedness)
+        be considered lossy.
+
+    """
+    def n_significant_bits(t):
+        dtypeobj = np.dtype(t)
+        kind = dtypeobj.kind
+        if kind == 'f':
+            # IEEE floating point point has an implied 1, increasing the number
+            # of significant bits given a mantisa of a certain size
+            return np.finfo(dtypeobj).nmant + 1
+        elif kind == 'b':
+            return 1
+        elif kind == 'u':
+            # 8 bits per byte
+            return dtypeobj.itemsize * 8
+        elif kind == 'i':
+            # 1 bit used for sign
+            return dtypeobj.itemsize * 8 - 1
+        else:
+            raise ValueError('Unknown image dtype.')
+
+    kind_in = np.dtype(dtypeobj_in).kind
+    kind_out = np.dtype(dtypeobj_out).kind
+    significant_bits_in = n_significant_bits(dtypeobj_in)
+    significant_bits_out = n_significant_bits(dtypeobj_out)
+
+    if kind_in == 'f' and kind_out != 'f':
+        has_loss = True
+    elif significant_bits_in > significant_bits_out:
+        has_loss = True
+    else:
+        has_loss = False
+
+    if has_loss:
+        warn("Possible precision loss when converting from {} to {}"
+             .format(dtypeobj_in, dtypeobj_out))
+
+    return has_loss
+
+
+def dtype_limits(image, clip_negative=None):
```

Comment:
This is an API change, isn't it?

Reply:
No I read it changed from ` clip_negative=False)` to ` clip_negative=None)` The original code is `=False` as far as I can tell.